src/main/java/net/pms/util/FileUtil.java

```diff
@@ -668,37 +669,55 @@ public boolean accept(File dir, String name) {
 		return found;
 	}
 
+	private static String externalSubsLang;
```

Comment:
@valib I don't understand why you made this variable. It seems to me to simply return the same result as the last call to getFileCharset. You may have made this for efficiency, but you can't do it like this in a multithreaded application, especially since the methods are static. To avoid making the test multiple times, you must keep the result in a variable in the calling method instead.

Another thread could have changed the value of ```externalSubsLang``` since the last time ```getFileCharset``` was called from the thread in question. It's not treadsafe either, so you risk getting garbage back.

I'm working on #689 and even though it's not related to that PR; I'm going to remove it there as it has other changes to ```FileUtil```.